### PR TITLE
New documentation site!

### PR DESCRIPTION
**Pull request by mroderick**
#### TL;DR
I would like to create a new documentation site for Sinon.JS. I was hoping we could have this done for 2.0, but then someone merged backwards incompatible changes into `master` ;-)

This PR is the first step towards a new site, importing and tidying up the existing `sinonjs/sinon-docs` as Markdown files. We don't have to merge it in it's current state, but it would be easier to recruit contributors to the effort, if it's in the `master` branch.

##### High barrier to entry

Current docs are difficult for casual contributors to contribute to.

Solution: use GitHub Flavoured Markdown, or something very close to it

##### Source changes without documentation updates

Contributors (repository owners included) often neglect updating the documentation when source changes.

Solution: keep documentation sources in the same repository as source code, so contributors can update them in same PR as source code changes.

##### Human interaction required for documentation site updates

Currently, only @cjohansen can deploy a new version of the documentation site.

Solution: fully automate build and deployment of documentation site using [metalsmith.io]( http://www.metalsmith.io) or [jekyll](http://jekyllrb.com) via heroku (or similar). Every new tag in the git repository should trigger rebuild and deploy of the site without any reliance on humans.

##### Documentation only covers current version

We should provide documentation for each tagged version (going forwards), it isn't that difficult, and makes it a lot easier for users to quickly find correct information. I've outlined the process in `docs/README.md`.

##### Misc.

* Embed the changelog to make it easier to follow changes
* Build some twitter automation to announce new versions
* Provide more examples of how to use Sinon.JS
* Provide a links page with links to related resources

### CODE REVIEW

**Comment by mantoni** [comment]
Thanks for pushing this forward. I haven't worked it through yet, but this seems to be a nice article on how to get things started with Metalsmith: https://segment.com/blog/building-technical-documentation-with-metalsmith/

**Comment by cjohansen** [comment]
Wow, that's a massive piece of work. Looks very solid to me. I'm merging, as I don't want to see this rot in a PR that never makes it.

**Comment by mroderick** [comment]
Cheers, I'll get on the next steps now
